frontend/src/views/admin/DashboardPage/DashboardPage.tsx

```diff
@@ -28,37 +49,106 @@ export const AdminDashboardPage = () => {
     }
   }, [isNotAllowed, isUserLoading]);
 
+  useEffect(() => {
+    if (!config.allowSignUp) {
+      setSignUpMode("disabled");
+      return;
+    }
+    if (config.inviteOnlySignUp) {
+      setSignUpMode("invite-only");
+    } else {
+      setSignUpMode("anyone");
+    }
+
+    if (config.allowSpecificDomainSignUp) {
+      setAllowSpecificDomain(config.allowSpecificDomainSignUp);
+    }
+  }, [config]);
+
+  const handleSubmit = async () => {
+    try {
+      config.allowSignUp = signUpMode !== "disabled";
```

Comment:
Avoid mutating the state directly. Create a new copy of config and mutate that. 

The updateServerConfig automatically fetches the latest config on change

https://dev.to/il3ven/common-error-accidentally-mutating-state-in-react-4ndg

Reply:
fixed